test/functional/cmdLineTests/shareClassTests/SCCMLTests/ShareClassesCMLTests-3.xml

```diff
@@ -1621,7 +1621,359 @@
+	</test>
+
+	<test id="Test 182: Test the sub-option Layer= with an incorrect layer number" timeout="600" runPath=".">
+		<command>$JAVA_EXE$ $currentMode$,layer=3 -version</command>
+		<output type="success" caseSensitive="yes" regex="no">JVMSHRC844E Invalid layer number 3</output>
+		<output type="required" caseSensitive="yes" regex="no">The existing top layer is 1</output>
+		<output type="required" caseSensitive="yes" regex="no">The new layer number should be 2</output>
+
+		<output type="failure" caseSensitive="no" regex="no">Unhandled Exception</output>
+		<output type="failure" caseSensitive="yes" regex="no">Exception:</output>
+		<output type="failure" caseSensitive="no" regex="no">corrupt</output>
+		<output type="failure" caseSensitive="yes" regex="no">Processing dump event</output>
+	</test>
+
+	<exec command="rm -f testjavacore.txt" quiet="false"/>	
+
+	<!-- Generate the javacore file here in order to test it on test 185-->
+	<test id="Test 183: Test the sub-option Layer= with a correct layer number and generate the javacore file" timeout="600" runPath=".">
+		<command>$JAVA_EXE$ -Xdump:java:events=vmstop,file=testjavacore.txt $currentMode$,layer=2 $CP_HANOI$ $PROGRAM_HANOI$</command>
+		<output type="success" caseSensitive="yes" regex="no">Puzzle solved!</output>
+		<output type="required" caseSensitive="yes" regex="no">JVMDUMP010I Java dump written to</output>
+		<output type="required" caseSensitive="yes" regex="no">JVMDUMP013I Processed dump event "vmstop"</output>
+		
+		<output type="failure" caseSensitive="yes" regex="no">JVMDUMP030W Cannot write dump to file</output>
+		<output type="failure" caseSensitive="no" regex="no">Unhandled Exception</output>
+		<output type="failure" caseSensitive="yes" regex="no">Exception:</output>
+		<output type="failure" caseSensitive="no" regex="no">corrupt</output>
+	</test>
+
+	<test id="Test 184: Verify that layer information is listed in the javacore">
+		<command>cat testjavacore.txt</command>				
+		<output type="success" caseSensitive="no" regex="no">1SCLTEXTCSTL   Cache Statistics for Top Layer</output>
+		<output type="required" caseSensitive="no" regex="yes" javaUtilPattern="yes">2SCLTEXTCLY[\s]*Cache layer[\s]*= 2</output>
+		<output type="required" caseSensitive="no" regex="no">1SCLTEXTCSAL   Cache Statistics for All Layers</output>
+
+		<output type="failure" caseSensitive="no" regex="no">No such file or directory</output>
+	</test>
+	
+	<exec command="rm -f testjavacore.txt" quiet="false"/>
+
+	<test id="Test 185: Check the number of Layers" timeout="600" runPath=".">
+		<command>$JAVA_EXE$ $currentMode$,listAllCaches</command>
+		<output type="success" caseSensitive="yes" regex="yes" javaUtilPattern="yes">ShareClassesCMLTests[\s]*Java(.)*(yes|persistent)(.)*(cr|non-cr|default)[\s]*0</output>
+		<output type="required" caseSensitive="yes" regex="yes" javaUtilPattern="yes">ShareClassesCMLTests[\s]*Java(.)*(yes|persistent)(.)*(cr|non-cr|default)[\s]*1</output>
```

Comment:
Why is it `(yes|persistent)`, shouldn't it be `(non-persistent|persistent)` ?

Reply:
You can also check that it should spaces after the layer number.

Reply:
Oh it is because I saw some other tests were using`` (yes|persistent)`` and the ``cache-type`` field seems to be ``(persistent|cache-type)`` for the previous tests.  I will change it if it is no longer using the ``persistent`` to indicate that field.